### ISSUE TITLE
test_build_hello_world_with_remote_embedded_tool_targets failing in 0.11.0rc3

### ISSUE THREAD

**Issue by katre** [bazelbuild/bazel#4627]
https://ci.bazel.build/blue/organizations/jenkins/Global%2Fbazel-tests/detail/bazel-tests/500/tests

Test is failing on the platforms:
- darwin-x86_64

Error:
```
-- Test log: -----------------------------------------------------------
INFO: $TEST_TMPDIR defined: output root default is '/private/var/tmp/_bazel_ci/14a274db1e6628a41b59a0d9bb823350/bazel-sandbox/1129898927531165150/execroot/io_bazel/_tmp/acd19529a89f4754e9dc4c27949ac850' and max_idle_secs default is '15'.
Loading:
Loading: 0 packages loaded
Analyzing: target //java/main:main_deploy.jar (7 packages loaded)
Analyzing: target //java/main:main_deploy.jar (7 packages loaded)
INFO: Analysed target //java/main:main_deploy.jar (20 packages loaded).
INFO: Found 1 target...
[0 / 2] [-----] BazelWorkspaceStatusAction stable-status.txt
ERROR: /private/var/tmp/_bazel_ci/14a274db1e6628a41b59a0d9bb823350/bazel-sandbox/1129898927531165150/execroot/io_bazel/_tmp/acd19529a89f4754e9dc4c27949ac850/root/d1885538425ba3a1c3a06bc153f9b8c6/external/bazel_tools/src/tools/singlejar/BUILD:291:1: C++ compilation of rule '@bazel_tools//src/tools/singlejar:input_jar' failed: I/O exception during sandboxed execution: com.google.devtools.build.lib.shell.ExecFailedException: java.io.IOException: Cannot run program "/private/var/tmp/_bazel_ci/14a274db1e6628a41b59a0d9bb823350/bazel-sandbox/1129898927531165150/execroot/io_bazel/_tmp/acd19529a89f4754e9dc4c27949ac850/root/d1885538425ba3a1c3a06bc153f9b8c6/execroot/__main__/_bin/xcode-locator": error=2, No such file or directory
Target //java/main:main_deploy.jar failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 5.433s, Critical Path: 0.02s
FAILED: Build did NOT complete successfully
------------------------------------------------------------------------
test_build_hello_world_with_remote_embedded_tool_targets FAILED: build failed .
/private/var/tmp/_bazel_ci/14a274db1e6628a41b59a0d9bb823350/bazel-sandbox/1129898927531165150/execroot/io_bazel/bazel-out/darwin-fastbuild/bin/src/test/shell/bazel/bazel_java_test.runfiles/io_bazel/src/test/shell/bazel/bazel_java_test:217: in call to test_build_hello_world_with_remote_embedded_tool_targets
INFO[bazel_java_test 2018-02-12 16:32:18 (+0000)] Cleaning up workspace
```

**Comment by katre** [comment]
Blocking release https://github.com/bazelbuild/bazel/issues/3959

**Comment by katre** [comment]
Failure is repeatable, but the entire test must be run with the RC, as so:
```
$ bazel build //src:bazel
...
INFO: Build completed successfully, 1 total action
jcater@jcater:~/repos/bazel:release-0.11.0$ cp bazel-bin/src/bazel /tmp/bazel-0.11.0rc3
jcater@jcater:~/repos/bazel:release-0.11.0$ /tmp/bazel-0.11.0rc3 test //src/test/shell/integration:run_test
....
INFO: Build completed, 1 test FAILED, 1244 total actions
//src/test/shell/integration:run_test                                    FAILED in 1 out of 3 in 76.5s
  Stats over 3 runs: max = 76.5s, min = 17.0s, avg = 36.9s, dev = 28.0s
    ERROR   .test_interrupt_kills_child (61.5s)

Executed 1 out of 1 test: 1 fails locally.
```

**Comment by katre** [comment]
This seems to be fixed in 0.11.0rc4.
